src/test/java/com/puppycrawl/tools/checkstyle/xpath/RootNodeTest.java

```diff
@@ -60,56 +61,64 @@ public void init() throws Exception {
     }
 
     @Test
     public void testGetLocalPart() {
-        assertEquals("Invalid local part", "ROOT", rootNode.getLocalPart());
+        assertEquals("ROOT", rootNode.getLocalPart(), "Invalid local part");
     }
 
     @Test
     public void testGetStringValue() {
-        assertEquals("Invalid string value", "ROOT", rootNode.getStringValue());
+        assertEquals("ROOT", rootNode.getStringValue(), "Invalid string value");
     }
 
     @Test
     public void testIterate() {
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE, rootNode.iterateAxis(AxisInfo.FOLLOWING));
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE,
-                rootNode.iterateAxis(AxisInfo.FOLLOWING_SIBLING));
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE, rootNode.iterateAxis(AxisInfo.PRECEDING));
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE,
-                rootNode.iterateAxis(AxisInfo.PRECEDING_SIBLING));
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE, rootNode.iterateAxis(AxisInfo.PARENT));
-        assertEquals("Result iterator does not match expected",
-                EmptyIterator.OfNodes.THE_INSTANCE, rootNode.iterateAxis(AxisInfo.PARENT,
-                        null));
+        try (AxisIterator following = rootNode.iterateAxis(AxisInfo.FOLLOWING)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, following,
+                    "Result iterator does not match expected");
+        }
+        try (AxisIterator followingSibling = rootNode.iterateAxis(AxisInfo.FOLLOWING_SIBLING)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, followingSibling,
+                    "Result iterator does not match expected");
+        }
+        try (AxisIterator preceding = rootNode.iterateAxis(AxisInfo.PRECEDING)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, preceding,
+                    "Result iterator does not match expected");
+        }
+        try (AxisIterator precedingSibling = rootNode.iterateAxis(AxisInfo.PRECEDING_SIBLING)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, precedingSibling,
+                    "Result iterator does not match expected");
+        }
+        try (AxisIterator parent = rootNode.iterateAxis(AxisInfo.PARENT)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, parent,
+                    "Result iterator does not match expected");
+        }
+        try (AxisIterator parentNull = rootNode.iterateAxis(AxisInfo.PARENT, null)) {
+            assertEquals(EmptyIterator.OfNodes.THE_INSTANCE, parentNull,
+                    "Result iterator does not match expected");
+        }
```

Comment:
again, here is a lot of duplication. May be change it to something like
`Stream.of(...).forEach(node -> try ....`
?

Reply:
it is better to postpone UT improvements to separate PR/issue.
So such PRs are technical migration.

Reply:
In addition, many tests can be rewritten as parameterized tests.
But this will greatly slow down the migration process. My goal is to move as quickly as possible to junit5. Then gradually add support for Google Truth and other modern tech.

Reply:
Before refactoring with parametrization, let's discuss BEFORE actual changes.